static/compiler-service.js

```diff
@@ -105,6 +105,12 @@ CompilerService.prototype.submit = function (request) {
                         case "abort":
                             error = "Request was aborted";
                             break;
+                        case "error":
+                            if (xhr.status === 504)
+                            {
+                                error = `Gateway timeout; returned statusCode: ${xhr.status}`
```

Comment:
We target ES5 for the client-side code, so this kind of string is disallowed there. A string concatenation is needed :)

Reply:
What @RabsRincon says is correct unfortunately! We may one day transcode ES6 to ES5, but right now our client code is ES5 to support the widest selection of desktop browsers.

Reply:
As above (had to put another comment to convince GitHub after I got confused)